.github/workflows/dev-release.yml

```diff
@@ -0,0 +1,141 @@
+name: Dev Build
+on:
+  workflow_dispatch:
+    inputs:
+      k6_version:
+        description: 'The version of the release, it must use the semantic versioning format with the v prefix. It is a development release so it is suggested to append a build metadata (e.g. v0.38.0-dev).'
+        required: true
+      k6_branch_or_commit:
+        description: 'k6 branch, tag or commit'
+        default: 'master'
+        required: true
+      go_version:
+        description: 'Go version for building binaries'
+        default: '1.17.x'
+        required: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      VERSION: ${{ github.event.inputs.k6_version }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.inputs.k6_branch_or_commit }}
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ github.event.inputs.go_version }}
+      - name: Install package builders
+        env:
+          GO111MODULE: 'off'
+        run: |
+          gopath="$(go env GOPATH)"
+          go get github.com/Masterminds/glide
+          go get -d github.com/mh-cbon/go-bin-deb \
+              && (cd "$gopath/src/github.com/mh-cbon/go-bin-deb" \
+              && glide install \
+              && go install)
+          go get -d github.com/mh-cbon/go-bin-rpm \
+              && (cd "$gopath/src/github.com/mh-cbon/go-bin-rpm" \
+              && glide install \
+              && go install)
+          sudo apt-get update -y
+          sudo apt-get install -y fakeroot rpm
+      - name: Generate Windows binary metadata
+        run: |
+          (cd && GO111MODULE=on go get github.com/josephspurrier/goversioninfo/cmd/goversioninfo@v1.2.0)
+          IFS=. read -a version <<< "$(echo $VERSION | sed 's:[^0-9\.]::g')"
+          # Need a blank versioninfo.json for the CLI overrides to work.
+          echo '{}' > versioninfo.json
+          goversioninfo -64 \
+            -platform-specific=true \
+            -charset="1200" \
+            -company="Grafana Labs AB" \
+            -copyright="© Grafana Labs AB. Licensed under AGPL." \
+            -description="A modern load testing tool, using Go and JavaScript" \
+            -icon=packaging/k6.ico \
+            -internal-name="k6" \
+            -original-name="k6.exe" \
+            -product-name="k6" \
+            -translation="0x0409" \
+            -product-version="${VERSION#v}"
```

Comment:
I think we need the `-ver-major`, `-ver-minor`, `-ver-patch` flags here, in addition to `-product-version`. Otherwise the version won't be shown properly in the Properties AFAIR. [That's what we do in the main workflow](https://github.com/grafana/k6/blob/59a8883676edf4e28edb39561c33ebd941e05944/.github/workflows/all.yml#L191-L194).

If these are not needed, then the array created on line 54 is not needed either.